superset-frontend/src/components/DatabaseSelector/changeDataBase/useChangeDataBase.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useRef } from 'react';
+import { FetchSchemas } from '../fetchSchemas';
+import { OnSelectChange } from '../onSelectChange';
+import { factoryChangeDataBase } from './factoryChangeDataBase';
+
+export const useChangeDataBase = ({
+  setSchemaOptions,
+  fetchSchemas,
+  onSchemaChange,
+  onDbChange,
+  onSelectChange,
+}: {
+  setSchemaOptions: (options: any[]) => void;
+  fetchSchemas: React.MutableRefObject<FetchSchemas>;
+  onSchemaChange?: (arg0?: any) => {};
+  onDbChange?: (db: any) => void;
+  onSelectChange: React.MutableRefObject<OnSelectChange>;
+}) => {
+  const changeDataBase = useRef(
+    factoryChangeDataBase({
+      setSchemaOptions,
+      fetchSchemas,
+      onSchemaChange,
+      onDbChange,
+      onSelectChange,
+    }),
+  );
+  useEffect(() => {
```

Comment:
Could this be implemented with a `useCallback` instead, moving the contents of `factoryChangeDataBase` into here? This is a lot of ref usage, which gets confusing and doesn't seem very idiomatic to hooks.

Reply:
`useCallback` does not tackle the problem the way I wanted it. It will generate a new function every time a parameter changes and will cause a new render.
As there are several functions that depend on each other, several rendes will be caused by this. When the `current` value is updated, a render is not trigged (which is why I don't return the `current` either).

The idea there is to behave like a class property. I want to change the value of these functions without triggering a new render. And of course, from the moment that this value is changed, everyone who uses this function (and has received these props in the past) will access the new value without a new rendering.

The combination of `useEffect` + `useRef` allows me to do this. Using `useCallback` or `useMemo` does not do this.